testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -6188,6 +6188,65 @@ void checkRegexpExtract(SqlOperatorFixture f0, FunctionAlias functionAlias) {
     f.checkNull("log(10, cast(null as real))");
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6224">[CALCITE-6224]
+   * Add LOG@ function (enabled in MYSQL, Spark library)</a>. */
+  @Test void testLog2Func() {
+    final SqlOperatorFixture f0 = fixture();
+    final Consumer<SqlOperatorFixture> consumer = f -> {
+      f.setFor(SqlLibraryOperators.LOG2);
+      f.checkScalarApprox("log2(2)", "DOUBLE NOT NULL",
+          isWithin(1.0, 0.000001));
+      f.checkScalarApprox("log2(4)", "DOUBLE NOT NULL",
+          isWithin(2.0, 0.000001));
+      f.checkScalarApprox("log2(65536)", "DOUBLE NOT NULL",
+          isWithin(16.0, 0.000001));
+      f.checkScalarApprox("log2(-2)", "DOUBLE NOT NULL",
+          "NaN");
+      f.checkScalarApprox("log2(2/3)", "DOUBLE NOT NULL",
+          "-Infinity");
+      f.checkScalarApprox("log2(2.2)", "DOUBLE NOT NULL",
+          "1.1375035237499351");
+      f.checkScalarApprox("log2(0.5)", "DOUBLE NOT NULL",
+          "-1.0");
+      f.checkScalarApprox("log2(3)", "DOUBLE NOT NULL",
+          isWithin(1.5849625007211563, 0.000001));
+      f.checkNull("log2(cast(null as real))");
+    };
+    f0.forEachLibrary(list(SqlLibrary.MYSQL, SqlLibrary.SPARK), consumer);
+  }
+
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6232">[CALCITE-6232]
+   * Using fractions in LOG function does not return correct results</a>. */
+  @Test void testLogFuncByConvert() {
+    // The fractional conversion of the Log function is reserved only for integer bits
+    final SqlOperatorFixture f0 = Fixtures.forOperators(true);
+    f0.setFor(SqlLibraryOperators.LOG, VmName.EXPAND);
+    final SqlOperatorFixture f = f0.withLibrary(SqlLibrary.BIG_QUERY);
+    f.checkScalarApprox("log(2/3, 2)", "DOUBLE NOT NULL",
+          "-Infinity");
+    f.checkScalarApprox("log(0,2)", "DOUBLE NOT NULL",
+          "-Infinity");
+    f.checkScalarApprox("log(1,2)", "DOUBLE NOT NULL",
+        "0.0");
+    f.checkScalarApprox("log(4/3,2)", "DOUBLE NOT NULL",
+        "0.0");
+    f.checkScalarApprox("log(5/3,2)", "DOUBLE NOT NULL",
+        "0.0");
+    f.checkScalarApprox("log(2/3, 10)", "DOUBLE NOT NULL",
+          "-Infinity");
+    f.checkScalarApprox("log(0,10)", "DOUBLE NOT NULL",
+          "-Infinity");
+    f.checkScalarApprox("log(1,10)", "DOUBLE NOT NULL",
+        "0.0");
+    f.checkScalarApprox("log(4/3,10)", "DOUBLE NOT NULL",
```

Comment:
These are good tests but I don't know that we should merge this until the pretty glaring bug is fixed

Reply:
I'm trying to fix it, but I'm sure it won't last long. I think we have to be honest and tell the user about this bug. Now maybe our difficulty is how to tell the user about this bug, should I add a TODO tag

Reply:
Why don't we fix the bug? Do you have an updated JIRA case for it?

Reply:
ok, this may take a while for me because i am learning this

Reply:
> Why don't we fix the bug? Do you have an updated JIRA case for it?

I don't know if this is true. https://issues.apache.org/jira/browse/CALCITE-6232

Reply:
What happens if you use the decimal form like `.5` instead of `1/2`?

Reply:
The result looks correct, it's only different when it comes to fractions